Sources/AblyAssetTrackingInternal/WorkerQueue/WorkerQueue.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// The WorkerQueue is responsible for enqueueing ``Worker``s and executing them.
+class WorkerQueue<PropertiesType, WorkerSpecificationType> where PropertiesType: WorkerQueueProperties {
```

Comment:
Is there anything we should be documenting about the `PropertiesType` parameter? In Android, the worker queue makes sure to [always pass a fresh instance to each worker](https://github.com/ably/ably-asset-tracking-android/blob/1a1869ef205280ca520e97f28d0b2733231f372e/common/src/main/java/com/ably/tracking/common/workerqueue/WorkerQueue.kt#L56), presumably to reduce the risk of shared mutable state. It would be good for us to have a similar way of reducing this risk here. Perhaps we could document that `PropertiesType` must have value semantics?

Reply:
An alternative approach would be to make `WorkerQueueProperties` conform to `NSCopying`, and just make sure we're also using a copy of properties in `worker.doWork`... But if we're going to stick with using structs when implementing properties, then value semantics handles this for us, so I'll stick to your suggestion.

e6022508208acb5950fcb28f342b52bd10382803